docs/designs/build-output.md

```diff
@@ -0,0 +1,127 @@
+# Build Output
+
+This document specifies docfx output file layout. It is designed to satisfy these requirements:
+
+- **Backward compatible with legacy system**: the output files should contain all the information needed to transform to whatever the *legacy system* desires. With an *undocumented* command line switch `--legacy`, `docfx` converts the output files to the legacy format.
+
+- **Dynamic build output maps directly to hosting server**: the build output for dynamic rendering should map directly to what the hosting layer expects, without permutation.
+  > *Hosting server* here is a fictional server using `documentdb`
+
+- **Efficient url lookup**: with an URL like `https://docs.microsoft.com/en-us/dotnet/api/system.string?view=netstandard-2.0#Instantiation`, it is efficient to lookup the content with proper locale fallback and moniker fallback.
+
+- **Static build output xcopy deployable**: should just xcopy static build output to any static hosting server.
+
+
+## URL Schema
+
+An **URL** is an universal identifier that confirms to [URL Standard](https://url.spec.whatwg.org/).
+
+For *dynamic rendering*, `docfx` produces an output for lookup by the following URL schema:
+
+`https://{host}/{locale?}/{site-url}/?view={moniker?}`
```

Comment:
why static and dynamic outputs are different?

Reply:
cannot have query string for static rendering. dynamic render has to follow docs.